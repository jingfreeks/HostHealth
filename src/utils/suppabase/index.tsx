export {default as supabase} from './suppabase';
