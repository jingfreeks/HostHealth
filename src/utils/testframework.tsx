import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
// import {QueryClient, QueryClientProvider} from '@tanstack/react-query';
// // import {DARK_THEME} from '@/utils/constants';
// // import {ThemeProvider} from 'styled-components/native';

type MockProviderType = {
  children?: React.ReactNode;
  store?: object;
};
export const MockProvider = (props: MockProviderType) => {
  const {store, children} = props;
  const mockStore = configureStore([thunk]);
  const mocked = mockStore(store);
  return <Provider store={mocked}>{children}</Provider>;
};

export const testingProps = (reference: string): unknown => ({
  testID: reference,
  accessibilityLabel: reference,
});

// const queryClient = new QueryClient({
//   defaultOptions: {
//     queries: {
//       retry: false,
//     },
//   },
// });
// export const wrapper = ({children}) => (
//   <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
// );
