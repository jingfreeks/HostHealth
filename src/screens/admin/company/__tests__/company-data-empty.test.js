import React from 'react'
import { fireEvent } from '@testing-library/react-native';
import {renderWithProviders} from '@/utils/testframeworknew';
import {Company} from '../index';


const deleteCompany=()=>jest.fn()
jest.mock('@/slice',()=>{
  return{
    useGetCompanyQuery:()=>({
      isLoading:false,
      isSuccess:true,
      isError:false,
      data:[]
    }),
    useDeleteCompanyMutation:()=>[deleteCompany,{isLoading:false,isError:false}],
  }
})
describe('Company admin empty screen', () => {
  it('Should work as expected to get snapshot', () => {
    const all = renderWithProviders(<Company />);
    expect(all.toJSON()).toMatchSnapshot();
  });

  it('Should to trigger Create Button', () => {
    const all = renderWithProviders(<Company />);
    const el=all.getByTestId('CompanyCreateButtonTestId')
    fireEvent(el,'onPress')
    expect(all.toJSON()).toBeTruthy();
  });
});
