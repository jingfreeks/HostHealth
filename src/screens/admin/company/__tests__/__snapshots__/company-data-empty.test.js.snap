// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company admin empty screen Should work as expected to get snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginBottom": 10,
      },
    ]
  }
>
  <RCTScrollView
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View />
  </RCTScrollView>
  <View
    collapsable={false}
    style={
      {
        "alignItems": "center",
        "borderRadius": 28,
        "bottom": 0,
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 16,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "transform": [
          {
            "scale": 1,
          },
        ],
      }
    }
  >
    <View
      style={
        [
          {
            "borderRadius": 3,
            "overflow": "hidden",
          },
          {
            "borderRadius": 3,
          },
          {
            "borderRadius": 28,
            "elevation": 4,
          },
          false,
        ]
      }
    >
      <View
        accessibilityLabel="CompanyCreateButtonTestId"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": false,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="CompanyCreateButtonTestId"
      >
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#ca71eb",
              "borderColor": "#2089dc",
              "borderRadius": 3,
              "borderWidth": 0,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#ffffff",
                "fontSize": 18,
                "marginHorizontal": 8,
                "paddingVertical": 1,
                "textAlign": "center",
              }
            }
          >
            Create
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
